File: SudokuKMP/shared/src/commonMain/kotlin/com/nextjedi/sudokustreak/model/SudokuCell.kt
package com.nextjedi.sudokustreak.model

data class SudokuCell(
    val value: Int = 0,
    val isInitial: Boolean = false,
    val notes: List<Int> = emptyList(),
    val isSolverFilled: Boolean = false
)

typealias SudokuGrid = List<List<SudokuCell>>

fun emptyGrid(): SudokuGrid = List(9) { List(9) { SudokuCell() } }

fun SudokuGrid.toMutableGrid(): Array<Array<SudokuCell>> =
    Array(9) { r -> Array(9) { c -> this[r][c] } }

fun Array<Array<SudokuCell>>.toGrid(): SudokuGrid =
    List(9) { r -> List(9) { c -> this[r][c] } }

fun SudokuGrid.toIntGrid(): Array<IntArray> =
    Array(9) { r -> IntArray(9) { c -> this[r][c].value } }

File: SudokuKMP/iosApp/iosApp/Models/SudokuCell.swift
import Foundation

struct SudokuCell {
    var value: Int
    var isInitial: Bool
    var isSolverFilled: Bool
    var notes: [Int]

    init(value: Int = 0, isInitial: Bool = false, isSolverFilled: Bool = false, notes: [Int] = []) {
        self.value = value
        self.isInitial = isInitial
        self.isSolverFilled = isSolverFilled
        self.notes = notes
    }
}

typealias SudokuGrid = [[SudokuCell]]

func emptyGrid() -> SudokuGrid {
    Array(repeating: Array(repeating: SudokuCell(), count: 9), count: 9)
}

func toIntGrid(_ grid: SudokuGrid) -> [[Int]] {
    grid.map { row in row.map { $0.value } }
}
